client/src/components/global/Date.js

```diff
@@ -0,0 +1,49 @@
+import React, { Component, PropTypes } from 'react';
+import format from 'date-fns/format';
+import parse from 'date-fns/parse';
+
+export default class Date extends Component {
+  static propTypes = {
+    prefix: PropTypes.string,
+    format: PropTypes.string,
+    date: PropTypes.oneOfType([
+      PropTypes.string,
+      PropTypes.object
+    ])
+  };
+
+  formatDate() {
+    let date = null;
+    if (this.props.date) {
+      if (this.props.date instanceof Object) {
+        date = (
+          format(this.props.date, this.props.format)
+        );
+      } else {
+        date = (
+          format(parse(this.props.date), this.props.format)
+        );
+      }
+    }
+
+    return date;
+  }
+
```

Comment:
No need to declare variables if you're just going to return them. Again, use guard clauses. Also, template strings are better than string concatenation in ES6. You can get this down to a two line method.

```
formatString(date) {
  if (!this.props.prefix || this.props.prefix.length === 0) return date;
  return `${this.props.prefix} ${date}`;
}
```